issues/9-margin-and-padding-on-base-component/PLAN.md

```diff
@@ -0,0 +1,140 @@
+# Implementation Plan: Margin and Padding on Base Component
+
+## Summary
+
+Add spacing utility props to the Cask component to enable configuration-based margins and padding. This enhancement will provide a consistent API for spacing control similar to design systems like MUI or Tailwind CSS, allowing developers to apply spacing with props like `ml='xl'`, `p='m'`, etc.
+
+## Scope
+
+### In Scope
+- Add margin props: `m`, `ml`, `mr`, `mt`, `mb`, `mx`, `my`
+- Add padding props: `p`, `pl`, `pr`, `pt`, `pb`, `px`, `py`
+- Define spacing scale CSS variables in `global.css`
+- Integrate with Next Yak styling system for CSS generation
+- Type-safe prop interfaces with auto-completion
+- Comprehensive unit tests for new functionality
+- Update component documentation
+
+### Out of Scope
+- Responsive spacing props (e.g., `ml={{ xs: 's', md: 'xl' }}`)
+- Negative margin values
+- Custom spacing values outside the defined scale
+- Integration with existing Card component (separate task)
+
+## File/Directory Map
+
+```
+src/
+├── app/
+│   └── globals.css                 # ADD: CSS spacing variables
+├── cask/
+│   ├── Cask.tsx                   # MODIFY: Add spacing props and styling
+│   ├── Cask.test.tsx              # MODIFY: Add tests for spacing functionality
+│   ├── spacing.ts                 # ADD: Spacing type definitions and utilities
+│   └── index.ts                   # MODIFY: Export spacing types if needed
+docs/
+└── ADR/
+    └── 0005-spacing-system.md      # ADD: Document spacing system decisions
+```
+
+## Acceptance Criteria
+
+1. **CSS Variables**: All spacing values defined in `global.css` as CSS custom properties
```

Comment:
Mention the values - i.e. copy them from the task. The plan.md is supposed to be a self sufficient description of the task - so all relevant information must be in there.